### DISCUSSION TITLE
How can I speed up integration?

### DISCUSSION THREAD

**Discussion by timoast** [satijalab/seurat#3999] [Q&A] [answered]
How can I speed up steps in the integration workflow when working with large datasets, or when integrating many different datasets?

**Comment by satijalab** [comment, marked as answer]
Working with multiple or large datasets can reduce the speed of the standard Seurat integration workflow. We offer three strategies, which can be combined, to assist users who wish to speed up these steps. First, Seurat allows for the use of reciprocal PCA (‘rpca’), as an alternative to canonical correlation analysis when identifying anchors, which substantially improves speed and memory. Second, when working with many datasets, users can specify one dataset (or a subset) as a baseline (or ‘reference’), which therefore avoids finding anchors between all pairs of datasets. Finally, anchor-finding between dataset pairs can be run in parallel by setting a parallelization plan (see the [future vignette](https://satijalab.org/seurat/future_vignette.html) for details). We describe these steps in a vignette, which also demonstrates how to integrate multiple datasets totaling >200,000 cells in ~30 minutes.

**Reply by warming151** [reply]
Hi! I wanna check on the  parallelization plan. However the link is expired. Would you pls provide a valid one? thanks!

**Reply by Priyanshi-Shah** [reply]
+1 , The link is not working for me either. Could you please share the updated version?

**Reply by ghost** [reply]
I cant use the link eighter

**Reply by lizchcase** [reply]
https://satijalab.org/seurat/archive/v3.0/future_vignette.html
